### PR TITLE
TNO-1447: Search by alias

### PR DESCRIPTION
**Pull request by jdtoombs**
Tried many variations of an elastic query to match aliases, given that this ticket is a "3 pointer" I settled with being selective about our aliases for now. This will at least eliminate the issue we were having before.

- my minister filter will now search by the aliases rather than the name 

I would like to eventually have it loop through the comma separated list and query each piece of text as an exact search; however, it was not as intuitive as I was hoping it was.

### CODE REVIEW

File: libs/net/dal/Migrations/1.0.43/Up/PreUp/00-Minister.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+DO $$
+BEGIN
+ALTER TABLE public."minister" ADD COLUMN "aliases" TEXT;
```
Comment (Fosol) [inline libs/net/dal/Migrations/1.0.43/Up/PreUp/00-Minister.sql:3]: I don't understand why you are doing this.  The DB migration should handle this.

File: libs/net/dal/Migrations/20230707180726_1.0.43.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+namespace TNO.DAL.Migrations
+{
+    /// <inheritdoc />
+    public partial class _1043 : SeedMigration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            PreUp(migrationBuilder);
+            migrationBuilder.AlterColumn<long>(
```
Comment (Fosol) [inline libs/net/dal/Migrations/20230707180726_1.0.43.cs:16]: I don't understand why there are so many changes to the Minister table?

File: libs/net/entities/Minister.cs [hunk 3]
```diff
@@ -13,6 +13,10 @@ public class Minister : BaseType<int>
     #region Constructors
     protected Minister() { }
 
+    /// <summary>
+    [Column("aliases")]
+    public string? Aliases { get; set; } = "";
```
Comment (Fosol) [inline libs/net/entities/Minister.cs:18]: Don't make it nullable.  Set the default value in the `MinisterConfiguration` to be an empty string.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: libs/net/dal/Migrations/20230707180726_1.0.43.cs (hunk 2 above)
Comment (jdtoombs) [reply libs/net/dal/Migrations/20230707180726_1.0.43.cs:16]: I'm not sure what happened here either, debugging it. I redid the migration after pulling in changes and it spat this out.

File: libs/net/dal/Migrations/1.0.43/Up/PreUp/00-Minister.sql (hunk 1 above)
Comment (jdtoombs) [reply libs/net/dal/Migrations/1.0.43/Up/PreUp/00-Minister.sql:3]: Forgot to remove this!

File: libs/net/dal/Migrations/20230707180726_1.0.43.cs (hunk 2 above)
Comment (Fosol) [reply libs/net/dal/Migrations/20230707180726_1.0.43.cs:16]: @CelloLiu @kylgarmor 

You need to correctly follow this process when removing invalid migrations that occur when another PR creates one.
Before getting the latest code and merging.
`make db-rollback n={prior version}`
`make db-remove`
Manually undo all changes to the `TNOContextModelSnapshot.cs` with git.
Get the latest code changes.
`make db-update`
`make db-add n={your new version}`
`make db-update`
Now test to make sure your rollback works
`make db-rollback n={prior version}`
`make db-update`

File: libs/net/dal/Configuration/MinisterConfiguration.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using TNO.Entities;
+
+public class MinisterConfiguration : BaseTypeConfiguration<Minister, int>
+{
+    public override void Configure(EntityTypeBuilder<Minister> builder)
+    {
+        builder.Property(m => m.Aliases).HasDefaultValueSql("''");
```
Comment (Fosol) [inline libs/net/dal/Configuration/MinisterConfiguration.cs:11]: Need to add a max length on strings too

**Review by Fosol — CHANGES_REQUESTED** [review]
Just one small change still needed

**Review by Fosol — CHANGES_REQUESTED** [review]
You'll need to update to `1.0.43`

**Comment by jdtoombs** [comment]
@Fosol - updated just so you know

**Review by Fosol — APPROVED** [review]

**Pull request merged**